Subject: [patch] uml fix some ptrace functions returns values.patch
From: Paolo 'Blaisorblade' Giarrusso <blaisorblade_spam@yahoo.it>
Patch-mainline: 2.6.11-rc1

This patch adds ptrace_setfpregs and makes these functions return -errno on
failure.

Signed-off-by: Paolo 'Blaisorblade' Giarrusso <blaisorblade_spam@yahoo.it>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Acked-by: Gerd Knorr <kraxel@suse.de>
---
 arch/um/sys-i386/ptrace_user.c |   19 ++++++++++++++++---
 1 files changed, 16 insertions(+), 3 deletions(-)

diff -puN arch/um/sys-i386/ptrace_user.c~uml-fix-some-ptrace-functions-returns-values arch/um/sys-i386/ptrace_user.c
--- 25/arch/um/sys-i386/ptrace_user.c~uml-fix-some-ptrace-functions-returns-values	Tue Nov 23 17:14:57 2004
+++ 25-akpm/arch/um/sys-i386/ptrace_user.c	Tue Nov 23 17:14:57 2004
@@ -17,17 +17,30 @@
 
 int ptrace_getregs(long pid, unsigned long *regs_out)
 {
-	return(ptrace(PTRACE_GETREGS, pid, 0, regs_out));
+	if (ptrace(PTRACE_GETREGS, pid, 0, regs_out) < 0)
+		return -errno;
+	return 0;
 }
 
 int ptrace_setregs(long pid, unsigned long *regs)
 {
-	return(ptrace(PTRACE_SETREGS, pid, 0, regs));
+	if (ptrace(PTRACE_SETREGS, pid, 0, regs) < 0)
+		return -errno;
+	return 0;
 }
 
 int ptrace_getfpregs(long pid, unsigned long *regs)
 {
-	return(ptrace(PTRACE_GETFPREGS, pid, 0, regs));
+	if (ptrace(PTRACE_GETFPREGS, pid, 0, regs) < 0)
+		return -errno;
+	return 0;
+}
+
+int ptrace_setfpregs(long pid, unsigned long *regs)
+{
+	if (ptrace(PTRACE_SETFPREGS, pid, 0, regs) < 0)
+		return -errno;
+	return 0;
 }
 
 static void write_debugregs(int pid, unsigned long *regs)
_
